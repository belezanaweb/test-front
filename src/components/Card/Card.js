import React from 'react'
import { createUseStyles } from 'react-jss'

const useStyles = createUseStyles({
  defaultCard: {
    width: '80%',
    minWidth: '320px',
    maxWidth: '600px',
    minHeight: '20px',
    borderRadius: '3px',
    backgroundColor: '#ffffff',
    boxShadow: '2px 2px 2px 1px rgba(0, 0, 0, 0.1)',
    border: 'none',
    padding: '10px',
    boxSizing: 'border-box',
    margin: '20px auto'
  }
})

const Card = (props) => {
  const { text, click } = props
  const classes = useStyles()

  return <div className={classes.defaultCard}>{props.children}</div>
}

export default Card
