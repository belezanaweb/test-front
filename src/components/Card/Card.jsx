import clsx from 'clsx';
import useStyles from './useStyles';

const Card = ({ children, className }) => {
  const classes = useStyles();

  return <div className={clsx(classes.root, className)}>{children}</div>;
};

export default Card;
