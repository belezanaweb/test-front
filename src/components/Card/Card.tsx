import styled from 'styled-components';

import Box from '../Box';

const Card = styled(Box)`
  background-color: #FFF;
  border-radius: 3px;
  box-shadow: 1px 1px 5px 0 rgba(0, 0, 29, 0.22);
  padding: 12px;
`;

export default Card;
