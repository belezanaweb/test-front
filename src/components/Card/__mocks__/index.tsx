import React from 'react';

const Card: React.FC = (): React.ReactElement => <div>card</div>;

export default Card;
