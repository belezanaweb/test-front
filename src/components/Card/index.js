import React from 'react';
import PropTypes from 'prop-types';
import { Title, Box, Container } from './style';

const Card = (props) => {
  const { title, heightBox, children } = props;

  return (
    <Container>
      <Title>
        {title}
      </Title>
      <Box height={heightBox}>
        {children}
      </Box>
    </Container>
  );
};

Card.propTypes = {
  title: PropTypes.string.isRequired,
  heightBox: PropTypes.number.isRequired,
  children: PropTypes.node.isRequired,
};

export default Card;
