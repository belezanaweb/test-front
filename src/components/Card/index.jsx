import React from 'react';
import CardStyled from './style';

const Card = props => (
  <CardStyled
    className="animated fadeIn"
    transparent={props.transparent && props.transparent}
  >
    {props.children && props.children}
  </CardStyled>
);

export default Card;
