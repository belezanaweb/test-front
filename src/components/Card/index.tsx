import { CardProps } from './interfaces'

import { Wrapper } from './styled'

const Card = ({ children }: CardProps) => {
  return <Wrapper>{children}</Wrapper>
}

export default Card
