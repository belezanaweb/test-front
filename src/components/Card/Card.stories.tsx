import React from 'react';

import theme, { Theming } from '../../theme';

import Box from '../Box';

import Card from './Card';

const Provider = (storyFn: () => React.ReactNode) => (
  <Theming theme={theme}>
    {storyFn()}
  </Theming>
);

export default {
  component: Card,
  decorators: [Provider],
  title: 'Card',
};

export const Default = () => (
  <Box css={{ padding: '20px' }}>
    <Card>&nbsp;</Card>
  </Box>
);
