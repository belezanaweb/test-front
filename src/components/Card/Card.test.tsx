import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import { provider } from '../../utils/testUtils';

import Card from './Card';

describe('<Card />', () => {
  it('should render Card', () => {
    const { container } = provider(<Card>Card</Card>, render);

    expect(container).toMatchSnapshot();
  });
});
