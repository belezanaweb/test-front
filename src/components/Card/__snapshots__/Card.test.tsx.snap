// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render Card 1`] = `
<div>
  <div
    class="sc-AxiKw sc-AxhCb jarWZH"
  >
    Card
  </div>
</div>
`;
